books/free-programming-books-langs.md

```diff
@@ -849,6 +850,11 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Learning graphqL](https://riptutorial.com/Download/graphql.pdf) - Compiled from StackOverflow Documentation (PDF)
 
 
+### Gremlin
+
+* [Practical Gremlin - An Apache TinkerPop Tutorial](https://www.kelvinlawrence.net/book/PracticalGremlin.pdf) - Kelvin R. Lawrence (PDF) - [HTML Version](https://www.kelvinlawrence.net/book/PracticalGremlin.html)
```

Comment:
only one link, please. choose the target that links to the other.
```suggestion
* [Practical Gremlin - An Apache TinkerPop Tutorial](https://www.kelvinlawrence.net/book/PracticalGremlin.html) - Kelvin R. Lawrence
```
It's usually not helpful to add a category for a single resource. Try to find another category, or see if there are other books in our lists that would fit into your category. Is Gremlin a GraphQL variant? Could the two be combined under "Graph Query Languages"? Are there other graph query languages?